py/bitbox02/bitbox02/bitbox02.py

```diff
@@ -108,6 +109,11 @@ def device_info(self) -> Dict[str, Any]:
             "monotonic_increments_remaining": response.device_info.monotonic_increments_remaining,
         }
 
+    def check_bitbox02_firmware_version(self) -> bool:
+        bitbox02_info = self.device_info()
```

Comment:
You need to use `self.version` and do the check in the constructor.

Reason:
`self.version` is set in the constructor by OP_INFO - `device_info()` might not be around in a future change. Actually we should remove the `version` field from there, it was added in the beginning but never used.

In the Go lib, the min bound check is done after pairing, as Reboot() needs to be available which runs over noise. In the Python lib, you could do both checks in the constructor and abort early with the right message (requires app/library upgrade, requires firmware upgrade).

Reply:
Updated:
The communication lib now checks that the firmware is newer than a minimum firmware version and raises an exeption if this is not the case.
    
Additionally a function is added in the bitbox02 to check whether the firmware is newer than what is required as a minimum, meaning that the library should probably be updated.